libs/text-splitters/langchain_text_splitters/html.py

```diff
@@ -33,149 +34,443 @@ class ElementType(TypedDict):
+                    for key, (content, _, dd) in active_headers.items()
+                    if node_dom_depth >= dd
+                }
+                documents.append(
+                    Document(page_content=node_content, metadata=header_meta)
+                )
+            else:
+                headers_to_remove = [
+                    key
+                    for key, (_, _, dd) in active_headers.items()
+                    if node_dom_depth < dd
+                ]
+                for key in headers_to_remove:
+                    del active_headers[key]
+                if node_content.strip():
+                    current_chunk.append(node_content)
+                    chunk_dom_depth = max(chunk_dom_depth, node_dom_depth)
+
+        sorted_nodes = sorted(nodes.items())
+        for _, node in sorted_nodes:
+            process_node(node)
+
+        self._finalize_chunk(current_chunk, active_headers, documents, chunk_dom_depth)
+        return documents
+
     def split_text_from_file(self, file: Any) -> List[Document]:
-        """Split HTML file.
+        """Split HTML content from a file into a list of Document objects.
 
         Args:
-            file: HTML file
+            file: A file path or a file-like object containing HTML content.
+
+        Returns:
+            A list of split Document objects.
         """
         try:
-            from lxml import etree
+            from bs4 import BeautifulSoup  # type: ignore[import-untyped]
         except ImportError as e:
             raise ImportError(
-                "Unable to import lxml, please install with `pip install lxml`."
+                "Unable to import BeautifulSoup/PageElement, \
+                    please install with `pip install \
+                    bs4`."
             ) from e
-        # use lxml library to parse html document and return xml ElementTree
-        # Explicitly encoding in utf-8 allows non-English
-        # html files to be processed without garbled characters
-        parser = etree.HTMLParser(encoding="utf-8")
-        tree = etree.parse(file, parser)
-
-        # document transformation for "structure-aware" chunking is handled with xsl.
-        # see comments in html_chunks_with_headers.xslt for more detailed information.
-        xslt_path = pathlib.Path(__file__).parent / "xsl/html_chunks_with_headers.xslt"
-        xslt_tree = etree.parse(xslt_path)
-        transform = etree.XSLT(xslt_tree)
-        result = transform(tree)
-        result_dom = etree.fromstring(str(result))
+        if isinstance(file, str):
+            with open(file, "r", encoding="utf-8") as f:
+                html_content = f.read()
+        else:
+            html_content = file.read()
 
-        # create filter and mapping for header metadata
-        header_filter = [header[0] for header in self.headers_to_split_on]
-        header_mapping = dict(self.headers_to_split_on)
+        soup = BeautifulSoup(html_content, "html.parser")
```

Comment:
perhaps move the parsing code that starts from in-memory content into a separate function, so that the parsing logic doesn't have to go through a file?